app/controllers/site_controller.rb

```diff
@@ -79,6 +45,11 @@ def coldwater
 
   private
 
+  def load_habitat;
+    @habitat = Habitat.where(name: action_name).first
```

Comment:
the `action_name` variable doesn't seem been defined anywhere

Reply:
It's a rails helper that returns the name of the current action